Source/WebCore/workers/service/background-fetch/BackgroundFetchRegistration.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(SERVICE_WORKER)
+
+#include "ActiveDOMObject.h"
+#include "BackgroundFetchFailureReason.h"
+#include "BackgroundFetchInformation.h"
+#include "BackgroundFetchResult.h"
+#include "EventTarget.h"
+#include "JSDOMPromiseDeferred.h"
+#include <wtf/RefCounted.h>
+#include <wtf/text/WTFString.h>
+
+namespace WebCore {
+
+struct BackgroundFetchInformation;
+class BackgroundFetchRecord;
+struct BackgroundFetchRecordInformation;
+struct CacheQueryOptions;
+class FetchRequest;
+
+class BackgroundFetchRegistration : public RefCounted<BackgroundFetchRegistration>, public EventTarget, public ActiveDOMObject {
+    WTF_MAKE_ISO_ALLOCATED(BackgroundFetchRegistration);
+public:
+    static Ref<BackgroundFetchRegistration> create(ScriptExecutionContext&, BackgroundFetchInformation&&);
+    ~BackgroundFetchRegistration();
+
+    String id() const { return m_information.identifier; }
+    uint64_t uploadTotal() const { return m_information.uploadTotal; }
+    uint64_t uploaded() const { return m_information.uploaded; }
+    uint64_t downloadTotal() const { return m_information.downloadTotal; }
+    uint64_t downloaded() const { return m_information.downloaded; }
+    BackgroundFetchResult result() { return m_information.result; }
+    BackgroundFetchFailureReason failureReason() { return m_information.failureReason; }
+    bool recordsAvailable() { return m_information.recordsAvailable; }
```

Comment:
Should be marked as const. Also, this would look much better with a prefix like "areRecordsAvailable".

Reply:
Added const, I kept recordsAvailable since it will probably only be used by binding code